src/kbmod/image_info.py

```diff
@@ -12,15 +17,33 @@ def __init__(self):
         self.wcs = None
         self.center = None
         self.obs_code = ""
+        self.filename = None
+        self.visit_id = None
 
-    def populate_from_fits_file(self, filename):
+    def populate_from_fits_file(self, filename, visit_in_filename=None):
         """Read the file stats information from a FITS file.
 
         Parameters
         ----------
         filename : string
             The path and name of the FITS file.
+        visit_in_filename : list of ints
+            A list containg the first last character of the visit ID
+            contained in the filename and the first character after the visit ID
+            (e.g. [0, 6] will use characters from 0 to 5 inclusive).
         """
+        # Skip non-FITs files.
+        if len(filename) < 5 or filename[-5:] != ".fits":
```

Comment:
5 is a magic number because I was checking that the file ended with the ".fits" suffix and ".fits" has 5 characters. So before I did the check, I wanted to check the string was long enough to even contain the suffix.

Do we want to support an extension like `.fits.tar.bz2`?

In the meantime, I changed this to your first suggestion.

Reply:
I think there are benefits to supporting extensions like `.bz2` because Astropy is able to lazy-load and decompress them on the fly. I'm not sure what the performance cost is for cases when we run through all the files to read headers only for example, but in previous space saving can be on the order of 3-4 times (SDSS for example has a nominal file size of 12MB which is cca 3MB bzipped). See this [note from SDSS](https://www.sdss4.org/dr16/imaging/images/#CorrectedFrames) for more details.